File: 0524-longest-word-in-dictionary-through-deleting/0524-longest-word-in-dictionary-through-deleting.java
public class Solution {
    public String findLongestWord(String s, List<String> a) {
        int n = a.size();
        int m = s.length();
        int[] idx = new int[n];
        int[] ct = new int[n];
        int mx = 0;
        List<String> x = new ArrayList<>();

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (i < s.length() && idx[j] < a.get(j).length() && s.charAt(i) == a.get(j).charAt(idx[j])) {
                    idx[j]++;
                }
                if (idx[j] == a.get(j).length()) {
                    mx = Math.max(mx, a.get(j).length());
                    x.add(a.get(j));
                }
            }
        }
        if (mx == 0) {
            return "";
        }

        String ans = String.format("%1$-" + 1005 + "s", "z").replace(' ', 'z');
        for (String i : x) {
            if (i.length() == mx) {
                ans = (ans.compareTo(i) < 0) ? ans : i;
            }
        }
        return ans;
    }
}

File: 0524-longest-word-in-dictionary-through-deleting/0524-longest-word-in-dictionary-through-deleting.cpp
class Solution {
public:
    string findLongestWord(string s, vector<string>& a) {
        int n = a.size(); 
        int m = s.size(); 
        vector<int>idx(n , 0 ) , ct(n , 0 ) ;
        int mx = 0 ;
        vector<string>x; 
        for(int i = 0; i < m ; i++ ){
            for(int j = 0; j < n ; j++){
                if(s[i] == a[j][idx[j]]){
                    idx[j]++; 
                }
                if( idx[j] == a[j].size()){
                    mx = max(mx , (int)a[j].size()); 
                    x.push_back(a[j]); 
                    continue; 
                }
            }
        }
        if(mx == 0 ){
            return  ""; 
        }
        //cout<<mx << endl; 
        string ans(1005 , 'z'); 
        for(auto i : x){
            if(i.size() == mx){
                ans = min(ans, i);
            }
        }
        return ans; 
    }
};